ast/expressions.go

```diff
@@ -87,7 +88,22 @@ func (n *ValueExpr) Format(w io.Writer) {
 	case types.KindMysqlDecimal:
 		s = n.GetMysqlDecimal().String()
 	case types.KindBinaryLiteral:
-		s = n.GetBinaryLiteral().ToString()
+		var sb *bytes.Buffer
+		// Hex literal has unsigned flag but bit literal not.
+		// See `types.DefaultTypeForValue`
+		if n.Type.Flag&mysql.UnsignedFlag != 0 {
+			sb = bytes.NewBufferString("x'")
```

Comment:
Use `BinaryLiteral.String()`.

Reply:
`String` doesn't suite here. `select "0xbcde"` has different field type from `select x'bcde'`. you can see it if you try it in MySQL cli with `--column-type-info`.

Reply:
OK, then this will do.
```
s = fmt.Sprintf("x'%x'", n.GetBytes())
```